packages/demand-deeplink/src/validate.ts

```diff
@@ -213,14 +234,30 @@ export function validate(deeplinkData: DeeplinkData): ValidationResponse {
   return errors.length ? { ok: false, errors } : { ok: true }
 }
 
-export function validateLegToQuotes(leg: JourneyLeg, defaultBookingType: BookingType, path: string) {
+export function validateLegToQuotes(
+  leg: JourneyLeg,
+  defaultBookingType: BookingType = BookingTypes.PREBOOK,
+  path: string,
+  { strict }: ValidationOptions = { strict: true }
+) {
   const errors = []
   const pickUpFields = excludeUndefined([leg.pickup, leg.pickupPlaceId, leg.pickupKpoi])
   const dropoffFields = excludeUndefined([leg.dropoff, leg.dropoffPlaceId, leg.dropoffKpoi])
   const pickupPosition = leg.pickupPosition
   const dropoffPosition = leg.dropoffPosition
   const bookingType = leg.bookingType ?? defaultBookingType
 
+  // ?? I don't get this logic  ??
+  //   why pickup or dropoff cannot be defined ?
+  //   error code means 'Pickup or dropoff are unacceptable parameters',
+  //
+  //   use case in web-booker
```

Comment:
usecase *

Reply:
There are multiple ways to provide pickup or dropoff info - 3 or 4 properties:
`const pickUpFields = excludeUndefined([leg.pickup, leg.pickupPlaceId, leg.pickupKpoi])`

`leg.pickup` is just string which is replaced under the hood to kpoi, placeId or lat/long, using AddressAutocomplete API. This is not allowed for quotes ATM